src/main/java/com/maxdemarzi/shortest/Service.java

```diff
@@ -438,6 +405,59 @@ private void streamShortestPathsUsingHandwrittenBFS(String centerEmail, List<Str
         }
     }
 
+    private static final Map<String, Integer> relationshipCosts = ImmutableMap.<String, Integer>builder()
+        .put("EqualTo", 1)
+        .put("AuthoredBy", 1)
+        .put("CoAuthorOf", 1)
+        .put("ContainsEmail", 1)
+        .put("Follows", 1)
+        .put("hasContact", 1)
+        .put("HasEmail", 1)
+        .put("HasUrl", 1)
+        .build();
+
+    private void streamShortestPathsUsingDijkstra(String centerEmail, List<String> bibEntries, List<String> edgeEmails, int maxLength, JsonGenerator jg) {
+        final Long centerNodeId;
+        try {
+            centerNodeId = nodeCache.getEmailNode(centerEmail);
+        } catch (ExecutionException e) {
+            return;
+        }
+        Collection<Long> startNodes = nodeCache.getBibliographEntryNodes(bibEntries);
```

Comment:
I think this is wrong, because we want to assume a path with a cost of 1 to the bibliography entries from the center node, not zero.